### DISCUSSION TITLE
Benchmarks

### DISCUSSION THREAD

**Discussion by fdrmrc** [fdrmrc/non_matching_test_suite#39] [Show and tell]
- Geometry: **piston**
- Number of elements original hexahedra: 365 712

The time considered here is the total time required to partition the original triangulation. Therefore, with the R-tree I sum the following steps:
1. Build the R-tree
2. Visit the nodes of a specific level and store the information about the agglomerates
3. Effectively agglomerate elements of the original triangulation together

With METIS, I record the time needed by the deal.II partitioner function. This function internally calls `METIS_PartGraphKway()` and flags the cells together.

![Screenshot 2023-12-14 at 11 34 38](https://github.com/fdrmrc/Polydeal/assets/50499769/a8eb3c0a-a25b-4903-8996-30046e475735)

- - - - - - - - - - -

- Geometry: **brain** (3D)
- Number of elements original hexahedra: 1 523 408

<img width="1078" alt="Screenshot 2023-12-14 at 18 50 00" src="https://github.com/fdrmrc/Polydeal/assets/50499769/d20594f0-5280-4ac4-9116-f259ded65f36">

**Comment by fdrmrc** [comment]
Another view of some agglomerates inside the piston mesh:

![immagine](https://github.com/fdrmrc/Polydeal/assets/50499769/1078c016-0201-421a-b381-f2ea7d137fa3)

**Reply by andreacangiani** [reply]
Very cool!! You say another view...do you have other nice pictures like this one?

**Reply by fdrmrc** [reply]
Sure, here they are (let me know if you prefer other views):
![immagine](https://github.com/fdrmrc/Polydeal/assets/50499769/fa700285-3ef0-4c8b-af82-e518975c09fb)

![immagine](https://github.com/fdrmrc/Polydeal/assets/50499769/7e6799cd-aaf4-40ba-8099-d1ef5d8fb542)

**Reply by fdrmrc** [reply]
Btw, within the next few days I plan to compute the metrics related to the quality of the agglomerates (taken from the "Neural Network paper").

**Reply by andreacangiani** [reply]
Fantastic, thanks! The timings above are striking!
